catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/JsonRpc.java

```diff
@@ -97,83 +100,126 @@ public void onClose(Session session, int statusCode, String reason) {
 
   @Override
   public void onError(Session session, Throwable ex) {
-    // no action required on error
+    if (ex instanceof SecureWebSocketException) {
+      SecurityLogger.audit("Received WebSockets request for user that is not logged in.");
+      handleMessage(
+          session,
+          ((SecureWebSocketException) ex).getWsMessage(),
+          (message, id) -> error(NOT_LOGGED_IN_ERROR, ex.getMessage()));
```

Comment:
Initially, I was going to do this because I think it's a good idea, but the lambda allows me to have access to the `ex` in order to do `ex.getMessage()`. If it's in a separate method, I can't do that.